pkg/controller/versions/strategy_v2_4.go

```diff
@@ -203,13 +203,13 @@ func (v *versionStrategyV2_4) validateExtensionProviders(spec *v2.ControlPlaneSp
 		return allErrors
 	}
 	for _, ext := range spec.MeshConfig.ExtensionProviders {
-		if ext.Prometheus == nil && ext.EnvoyExtAuthzHTTP == nil {
+		if ext.Prometheus == nil && ext.EnvoyExtAuthzHTTP == nil && ext.EnvoyExtAuthzGRPC == nil {
 			allErrors = append(allErrors, fmt.Errorf("extension provider '%s' does not define any provider - "+
-				"it must specify one of: prometheus or envoyExtAuthzHttp", ext.Name))
+				"it must specify one of: prometheus, envoyExtAuthzHttp, or envoyExtAuthzGrcp", ext.Name))
 		}
-		if ext.Prometheus != nil && ext.EnvoyExtAuthzHTTP != nil {
+		if ext.Prometheus != nil && ext.EnvoyExtAuthzHTTP != nil && ext.EnvoyExtAuthzGRPC == nil {
```

Comment:
This if statement doesn't check what it's supposed to check. Maybe you should count the number of fields that are set and then rewrite this and the previous statement so they simply check if `count == 0` or `count > 1`.